src/components/checkbox/bl-checkbox.ts

```diff
@@ -0,0 +1,82 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { live } from 'lit/directives/live.js';
+import { event, EventDispatcher } from '../../utilities/event';
+import '../icon/bl-icon';
+import style from './bl-checkbox.css';
+
+/**
+ * @tag bl-checkbox
+ * @summary Baklava Checkbox component
+ */
+@customElement('bl-checkbox')
+export default class BlCheckbox extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  /**
+   * Sets the checked state for checkbox
+   */
+  @property({ type: Boolean, reflect: true })
+  checked = false;
+
+  /**
+   * Sets the disabled state for checkbox
+   */
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /**
+   * Sets the indeterminate state for checkbox
+   */
+  @property({ type: Boolean, reflect: true })
+  indeterminate = false;
+
+  /**
+   * Fires whenever user change the value of the checkbox.
+   */
+  @event('bl-checkbox-change') private onChange: EventDispatcher<boolean>;
+
+  handleChange(event: CustomEvent) {
+    const target = event.target as HTMLInputElement;
+    this.checked = target.checked;
+    this.onChange(target.checked);
+    this.indeterminate = false;
+  }
+
+  update(changedProperties: Map<string, unknown>) {
+    super.update(changedProperties);
+    if (this.indeterminate && this.checked) {
+      this.checked = false;
+      this.requestUpdate('checked', true);
+    }
+  }
+
+  render(): TemplateResult {
+    let icon = '';
+    if (this.checked) icon = 'check';
+    if (this.indeterminate) icon = 'minus';
+
+    return html`
+      <label>
+        <input
+          type="checkbox"
+          name="checkbox"
+          .checked=${live(this.checked)}
+          ?disabled=${this.disabled}
+          .indeterminate=${this.indeterminate}
```

Comment:
Do we also need `live` here?

Reply:
I meant also having live for indeterminate state like `.indeterminate=${live(this.indeterminate)}`

Reply:
Let's continue with this.